app/web/components/Button/Button.tsx

```diff
@@ -30,16 +35,20 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
+type ButtonClasses = {
+  [key: string]: string;
+};
+
 //type generics required to allow component prop
 //see https://github.com/mui-org/material-ui/issues/15827
-export type AppButtonProps<
-  D extends ElementType = "button",
-  P = Record<string, unknown>
-> = ButtonProps<D, P> & {
-  loading?: boolean;
-};
+export type AppButtonProps<D extends ElementType = "button"> =
+  ButtonProps<D> & {
+    loading?: boolean;
+    onClick?: MouseEventHandler<HTMLButtonElement>; // Dynamic type for different component types
+    classes?: Partial<ButtonClasses>; // Use the flexible ButtonClasses type here
+  };
```

Comment:
This looks scary but it's mostly just making the types more generic. I think some type started being enforced in a package that wasn't before, some mui types changed, etc. Since this button is reusable, we want it to be more flexible.